src/language/languageUnderstanding.js

```diff
@@ -128,6 +128,24 @@ class languageUnderstanding extends commonService {
         })
         
     };     
+
+    /**
+     * Publishes app. App must be trained before publishing.
+     * @returns {Promise.<object>}
+     */
+    publish(body) {
+        
+        const operation = {
+            "path": "luis/api/v2.0/apps/" + this.appID + "/publish",
+            "method": "POST",
```

Comment:
Can you add the body parameters?

```javascript
            "parameters": [{
                "name": "versionId",
                "description": "Version of the app",
                "value": null,
                "required": true,
                "type": "inBody",
                "typeName": "string"
            }, {
                "name": "isStaging",
                "description": "Publish destination",
                "value": null,
                "required": false,
                "type": "inBody",
                "typeName": "boolean"
            }, {
                "name": "region",
                "description": "Publish location",
                "value": null,
                "required": false,
                "type": "inBody",
                "typeName": "string"
            }]
```

I don't know if the descriptions are okay, I guessed from the documentation.

Reply:
The azure regions are a set/closed list. Is there a way to represent the choices or should it just stay as a string? Region list is https://docs.microsoft.com/en-us/azure/cognitive-services/luis/manage-keys and is the first part of the host name so https://westus.api.cognitive.microsoft.com/luis/v2.0/apps/YOUR-APP-ID?subscription-key=YOUR-SUBSCRIPTION-KEY has a region of "westus".

Reply:
You can add `"options" = ["westus", "eastus2", "westcentralus", "westeurope", "southeastasia"];` to the`region` parameter and it will validate it automatically.